File: w7/genslides/backend/api/schemas/style.py
"""Style API schemas."""

from pydantic import BaseModel


class StyleCandidate(BaseModel):
    """Style candidate image information."""

    filename: str
    url: str


class GenerateStyleRequest(BaseModel):
    """Request model for generating style candidates."""

    prompt: str


class GenerateStyleResponse(BaseModel):
    """Response model for style generation."""

    candidates: list[StyleCandidate]
    generation_cost: float


class SelectStyleRequest(BaseModel):
    """Request model for selecting a style."""

    prompt: str
    selected_image: str


class StyleResponse(BaseModel):
    """Response model for style information."""

    prompt: str
    image: str
    image_url: str


class GetStyleResponse(BaseModel):
    """Response model for getting style."""

    has_style: bool
    style: StyleResponse | None


class SelectStyleResponse(BaseModel):
    """Response model for style selection."""

    success: bool
    style: StyleResponse

File: w7/genslides/frontend/src/api/style.ts
import { get, post, put } from './index';
import type {
  StyleResponse,
  GenerateStyleResponse,
  SelectStyleResponse,
} from '@/types';

export const styleApi = {
  getStyle: (slug: string) =>
    get<StyleResponse>(`/slides/${slug}/style`),

  generateStyleCandidates: (slug: string, prompt: string) =>
    post<GenerateStyleResponse>(`/slides/${slug}/style/generate`, { prompt }),

  selectStyle: (slug: string, prompt: string, selectedImage: string) =>
    put<SelectStyleResponse>(`/slides/${slug}/style`, {
      prompt,
      selected_image: selectedImage,
    }),

  getStyleImageUrl: (slug: string, filename: string) =>
    `/api/slides/${slug}/style/${filename}`,
};
